File: lib/core/api/track/dto/response_track_dto.dart
import 'package:json/json.dart';

@JsonCodable()
class ResponseTrackDto { 
  final String name;
  final String trackImage;
  final String track;

  ResponseTrackDto({required this.name, required this.trackImage, required this.track});

  ResponseTrackDto copyWith(String? name, String? trackImage, String? track) => ResponseTrackDto(name: name ?? this.name, trackImage: trackImage ?? this.trackImage, track: track ?? this.track);
}

File: Application/Track/DTO/ResponseTrackDto.cs
using MusicG.Application.Genre.Dto;
using MusicG.Application.User.DTO;

public class ResponseTrackDto {
    public int Id {get;set;}
    public String Name {get;set;}
    public String Track {get;set;}
    public String TrackImage {get;set;}
    public ResponseUserDto User { get; set; }
    public ResponseGenreDto Genre { get; set; }
}